### PR TITLE
[GEOT-7139] Migrate JDBC tests from JUnit3 to JUnit4

### PR DESCRIPTION
**Pull request by stefanuhrig**
[GEOT-7139](https://osgeo-org.atlassian.net/browse/GEOT-7139) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

JUnit4 offers some advantages over JUnit3 like
 - support for parallel test execution
 - more flexibility via annotations
 - assumptions for skipping tests

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by stefanuhrig** [comment]
``JDBCDataStoreAPIOnlineTest#testGetFeatureWriterConcurrency()`` might block for some providers as I removed a PostGIS specific skip check based on database version from the base class. Let's see... Might require some further adaption in the actual providers...

**Comment by stefanuhrig** [comment]
Will continue tomorrow...

**Comment by stefanuhrig** [comment]
Looks good to me. The number of executed online tests is identical to before for each plugin (it's even one more for GeoPackage because there was a test-method not starting with ``test``).

Some tests are reported as skipped now, because I introduced assumptions in tests of the style

```java
public void test() {
    if (!cond) return;
    // test code
}
```

which look now like this

```java
public void test() {
    assumeTrue(cond);
    // test code
}
```

File: modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCGeometryOnlineTest.java [hunk 1]
```diff
@@ -41,40 +45,49 @@ public abstract class JDBCGeometryOnlineTest extends JDBCTestSupport {
     @Override
     protected abstract JDBCGeometryTestSetup createTestSetup();
 
+    @Test
```
Comment (aaime) [inline modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCGeometryOnlineTest.java:48]: The ``@SuppressWarnings("PMD.JUnit4TestShouldUseTestAnnotation")`` annotation can be removed in this class too.

File: modules/plugin/jdbc/jdbc-oracle/src/test/java/org/geotools/data/oracle/sdo/SDOOnlineTest.java [hunk 2]
```diff
@@ -68,54 +75,65 @@ protected void disconnect() throws Exception {
         super.disconnect();
     }
 
+    @Test
     public final void testGType() throws SQLException {
-        assertEquals(2003, SDO.gType(fixture.rectangle));
+        Assert.assertEquals(2003, SDO.gType(fixture.rectangle));
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-oracle/src/test/java/org/geotools/data/oracle/sdo/SDOOnlineTest.java:80]: Why not sure a static import for this one as well?

**Review by aaime — CHANGES_REQUESTED** [review]
Wow, epic PR!

Looks good to me, found a couple of minor issues, see below.

File: modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCGeometryOnlineTest.java (hunk 1 above)
Comment (stefanuhrig) [reply modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCGeometryOnlineTest.java:48]: Done

File: modules/plugin/jdbc/jdbc-oracle/src/test/java/org/geotools/data/oracle/sdo/SDOOnlineTest.java (hunk 2 above)
Comment (stefanuhrig) [reply modules/plugin/jdbc/jdbc-oracle/src/test/java/org/geotools/data/oracle/sdo/SDOOnlineTest.java:80]: The static import would be shadowed by the `assertEquals()` member method (https://stackoverflow.com/questions/6595980/import-static-does-not-work-when-the-class-has-methods-with-the-same-name-as-the).

**Review by aaime — APPROVED** [review]

**Pull request merged**